### PR TITLE
Controller-created pods should also with name length limit 253

### LINKED ISSUE #84580: The name of pod created by daemonset will be truncated at 64 characters
**What happened**:
The name of pod created by daemonset will be truncated at 64 characters

**What you expected to happen**:
The name of pod should be correctly expanded within length of 253 characters as shared here 
https://kubernetes.io/docs/concepts/overview/working-with-objects/names/

**How to reproduce it (as minimally and precisely as possible)**:

Use following yaml could reproduce the issue:
```yaml
apiVersion: "apps/v1"
kind: DaemonSet
metadata:
  name: "cmk-reconcile-nodereport-ds-cis-dev1-k8s-master-0.novalocal"
  labels:
    app: workplace
spec:
  selector:
    matchLabels:
      app: workplace
  template:
    metadata:
      labels:
        app: workplace
    spec:
      containers:
      - name: pod1
        image: <image>
        command: ["/bin/sleep"]
        args: ["36000"]
        imagePullPolicy: IfNotPresent
        resources:
          requests:
            cpu: "100m"
            memory: 100Mi
          limits:
            cpu: "100"
            memory: 100Mi
```
Pod created with name like `cmk-reconcile-nodereport-ds-cis-dev1-k8s-master-0.novaloca7q4d2`. As you see it is truncated, and the original work `novalocal`'s last character is replaced there.

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`):
```bash
# kubectl version
Client Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.4", GitCommit:"67d2fcf276fcd9cf743ad4be9a9ef5828adc082f", GitTreeState:"clean", BuildDate:"2019-09-18T14:51:13Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"15", GitVersion:"v1.15.4", GitCommit:"67d2fcf276fcd9cf743ad4be9a9ef5828adc082f", GitTreeState:"clean", BuildDate:"2019-09-18T14:41:55Z", GoVersion:"go1.12.9", Compiler:"gc", Platform:"linux/amd64"}
```
- Cloud provider or hardware configuration:
- OS (e.g: `cat /etc/os-release`):
```bash
# cat /etc/os-release
NAME=Fedora
VERSION="29 (Twenty Nine)"
ID=fedora
VERSION_ID=29
PLATFORM_ID="platform:f29"
PRETTY_NAME="Fedora 29 (Twenty Nine)"
ANSI_COLOR="0;34"
CPE_NAME="cpe:/o:fedoraproject:fedora:29"
HOME_URL="https://fedoraproject.org/"
DOCUMENTATION_URL="https://docs.fedoraproject.org/en-US/fedora/f29/system-administrators-guide/"
SUPPORT_URL="https://fedoraproject.org/wiki/Communicating_and_getting_help"
BUG_REPORT_URL="https://bugzilla.redhat.com/"
REDHAT_BUGZILLA_PRODUCT="Fedora"
REDHAT_BUGZILLA_PRODUCT_VERSION=29
REDHAT_SUPPORT_PRODUCT="Fedora"
REDHAT_SUPPORT_PRODUCT_VERSION=29
PRIVACY_POLICY_URL="https://fedoraproject.org/wiki/Legal:PrivacyPolicy"
```
- Kernel (e.g. `uname -a`):
```
# uname -a
Linux cis-prod1-k8s-master-0.novalocal 4.14.114-1.wf30.x86_64 #1 SMP Mon May 6 15:14:16 EEST 2019 x86_64 x86_64 x86_64 GNU/Linux
```
- Install tools:
- Network plugin and version (if this is a network-related bug):
- Others:

### PR DESCRIPTION
**Pull request by clarklee92**
Signed-off-by: clarklee92 <clarklee1992@hotmail.com>

**What type of PR is this?**
/kind bug
/kind feature

**What this PR does / why we need it**:
Both individual pods and controller created pods should have a uniform maximum name length of 253   Ref: https://kubernetes.io/docs/concepts/overview/working-with-objects/names/

**Which issue(s) this PR fixes**:

Fixes #84580 

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/storage/names/generate.go [hunk 1]
```diff
@@ -41,9 +42,8 @@ var SimpleNameGenerator NameGenerator = simpleNameGenerator{}
 
 const (
 	// TODO: make this flexible for non-core resources with alternate naming rules.
-	maxNameLength          = 63
 	randomLength           = 5
-	maxGeneratedNameLength = maxNameLength - randomLength
+	maxGeneratedNameLength = validation.DNS1123SubdomainMaxLength - randomLength
```
Comment (liggitt) [inline staging/src/k8s.io/apiserver/pkg/storage/names/generate.go]: changing this globally for all consumers of this generator allows them to generate names that are invalid for some resources.

if we want a version of the name generator that allows longer names, make a `NewSimpleNameGenerator(maxNameLength int)` variant that individual rest resources can make use of with their specific max length

Comment (clarklee92) [reply staging/src/k8s.io/apiserver/pkg/storage/names/generate.go]: I agree with you, although it won't cause any problems at present. I'll just change it where the pod name is generated.

**Pull request closed without merging**